Clean/Utils/Vector.lean

```diff
@@ -75,44 +85,63 @@ def induct_push {motive : {n: ℕ} → Vector α n → Prop}
   | ⟨ .mk (a::as), h ⟩ =>
     have : as.length + 1 = n := by rw [←h, Array.size_toArray, List.length_cons]
     subst this
-    -- TODO this should be constructive, so that `motive` can return a `Sort u`
-    obtain ⟨ as', a', ih ⟩ := exists_push (xs := ⟨.mk (a :: as), rfl⟩)
+    obtain ⟨ as', a', ih ⟩ := to_push ⟨.mk (a :: as), rfl⟩
     have ih' : motive as' := induct_push nil push as'
     have h' := push _ a' ih'
     rwa [ih]
 
-@[simp]
-def init {n} (create: Fin n → α) : Vector α n :=
-  match n with
-  | 0 => #v[]
-  | k + 1 =>
-    (init (fun i : Fin k => create i)).push (create k)
+def finRange (n : ℕ) : Vector (Fin n) n :=
+  ⟨ .mk (List.finRange n), List.length_finRange n ⟩
 
-theorem cast_init {n} {create: Fin n → α} (h : n = m) :
-    init create = (init (n:=m) (fun i => create (i.cast h.symm))).cast h.symm := by
+def mapFinRange {n} (create: Fin n → α) : Vector α n := finRange n |>.map create
```

Comment:
as few refactors here. `natInit` is now called `mapRange`, and `init` is called `mapFinRange`. also, we no longer use a custom implementation for `mapFinRange` (the old definition was really cumbersome to work with due to adding a fin-cast in every iteration)